Remove non-standard `width()` and `height()` functions of HTMLDocument

Ahmad-S792 opened the pull request:
#### 9e4d6906a76b73fae8fa281245ec884929dbd6a5
<pre>
Remove non-standard `width()` and `height()` functions of HTMLDocument

<a href="https://bugs.webkit.org/show_bug.cgi?id=271190">https://bugs.webkit.org/show_bug.cgi?id=271190</a>

Reviewed by Ryosuke Niwa and Michael Catanzaro.

This patch is to remove remaining code for dropped `HTMLDocument.width / height`
attributes, which were dropped in 171244@main (via IDL changes) and later
removed in 179619@main (both in 2016 - via removal from IDL).

* Source/WebCore/html/HTMLDocument.cpp:
(HTMLDocument::width): Deleted
(HTMLDocument::height): Deleted
* Source/WebCore/html/HTMLDocument.h:
* Source/WebKitLegacy/mac/DOM/DOMHTMLDocument.mm:
(width):
(height):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLDocument.cpp:
(webkit_dom_html_document_get_width):
(webkit_dom_html_document_get_height):

Canonical link: <a href="https://commits.webkit.org/276488@main">https://commits.webkit.org/276488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/450f296ada47a5cebb9487ea00af30ca71afc857

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44650 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23736 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47109 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47303 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40655 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46952 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27770 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21125 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36721 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45226 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38432 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17779 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18233 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39574 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2697 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40862 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39860 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48939 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19604 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16171 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43664 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20934 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42413 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/9979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21265 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20603 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/bc9b1fc26221eeef22ea7fff9b467c065716e1b5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/26082

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2a7c4204b70a915c2dca274c4a8e3a09c914860a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/26082

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5cb57afd4c7c480b66071c86d1b3354962f0cd78. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/26082

annevk commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLDocument.cpp:
```diff
@@ -299,18 +299,14 @@ glong webkit_dom_html_document_get_width(WebKitDOMHTMLDocument* self)
     WebCore::JSMainThreadNullState state;
     g_return_val_if_fail(WEBKIT_DOM_IS_HTML_DOCUMENT(self), 0);
-    WebCore::HTMLDocument* item = WebKit::core(self);
-    glong result = item->width();
-    return result;
+    return 0;
```
This should just become return 0.

Ahmad-S792 replied:
Done in [1f1978a](https://github.com/WebKit/WebKit/commit/1f1978a816967eb52e39cb17d27c6b03c576b9a3).

annevk commented on Source/WebKitLegacy/mac/DOM/DOMHTMLDocument.mm:
```diff
@@ -60,14 +60,12 @@ - (DOMHTMLCollection *)scripts
 {
-    WebCore::JSMainThreadNullState state;
-    return IMPL->width();
+    return NULL;
```
This should also be return 0 I think. NULL is not an int.

Ahmad-S792 replied:
Done in [1f1978a](https://github.com/WebKit/WebKit/commit/1f1978a816967eb52e39cb17d27c6b03c576b9a3).

annevk reviewed (commented):
I would remove " (from WebKit and WebKitLegacy)" as it's inaccurate. These are still exposed in WebKitLegacy.

annevk commented:
Thanks! This looks good to me, but I'd prefer @rniwa / @cdumez to sign off and someone from the Linux ports for the API changes.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f35e335baf302c59e849cccd9e56dbdd859f83e7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/26082

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/450f296ada47a5cebb9487ea00af30ca71afc857. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/26082

aperezdc reviewed (commented):
The DOM API has been deprecated since WebKitGTK version 2.22.0 (September 2018, four and a half years ago), and has never been part of WPE. Building WebKitGTK against GTK4 disables the whole DOM API. At this point we would expect everybody to have moved on; we keep the library and its symbols around avoid old applications that haven't been updated from segfaulting at load time.

I think this change is okay for the GTK port, if @carlosgcampos agrees let's merge it.

carlosgcampos commented:
LGTM

Commit:
Remove non-standard `width()` and `height()` functions of HTMLDocument

https://bugs.webkit.org/show_bug.cgi?id=271190

Reviewed by Ryosuke Niwa and Michael Catanzaro.

This patch is to remove remaining code for dropped `HTMLDocument.width / height`
attributes, which were dropped in 171244@main (via IDL changes) and later
removed in 179619@main (both in 2016 - via removal from IDL).

* Source/WebCore/html/HTMLDocument.cpp:
(HTMLDocument::width): Deleted
(HTMLDocument::height): Deleted
* Source/WebCore/html/HTMLDocument.h:
* Source/WebKitLegacy/mac/DOM/DOMHTMLDocument.mm:
(width):
(height):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLDocument.cpp:
(webkit_dom_html_document_get_width):
(webkit_dom_html_document_get_height):

Canonical link: https://commits.webkit.org/276488@main
